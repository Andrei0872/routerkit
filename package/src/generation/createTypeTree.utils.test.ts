import {
  createValidRouteIdentifier,
  handleRoutesWithVariable,
  hasRouteVariable,
  validateIdentifierValue
} from './createTypeTree.utils';

describe('[generation] createTypeTree utils', () => {
  describe('hasRouteVariable', () => {
    test('should return false when empty object', () => {
      expect(hasRouteVariable({})).toBeFalsy();
    });

    test('should return false when object has not variables', () => {
      expect(
        hasRouteVariable({
          info: ['/', 'info'],
          location: ['/', 'location']
        })
      ).toBeFalsy();
    });

    test('should return false when object has not nested variables', () => {
      expect(
        hasRouteVariable({
          root: ['/'],
          location: { map: ['/', 'location', 'map'] }
        })
      ).toBeFalsy();
    });

    test('should return false when object has nested variable', () => {
      expect(
        hasRouteVariable({
          root: ['/'],
          location: { ':city': ['/', 'location', 'string'] }
        })
      ).toBeFalsy();
    });

    test('should return true when object has variable', () => {
      expect(
        hasRouteVariable({
          root: ['/'],
          ':city': ['/', 'string']
        })
      ).toBeTruthy();
    });
  });

  describe('handleRoutesWithVariable', () => {
    test('should return both routes without variable and variable data', () => {
      expect(
        handleRoutesWithVariable({
          info: ['/', 'info'],
          ':city': ['/', 'string']
        })
      ).toEqual({
        routesWithoutVariable: { info: ['/', 'info'] },
        variable: { name: 'city', value: ['/', 'string'] }
      });
    });

    test('should have empty routesWithoutVariable', () => {
      expect(
        handleRoutesWithVariable({
          ':city': ['/', 'string']
        })
      ).toEqual({
        routesWithoutVariable: {},
        variable: { name: 'city', value: ['/', 'string'] }
      });
    });
  });

  describe('createValidRouteIdentifier', () => {
    test('should match createIdentifier', () => {
      expect(createValidRouteIdentifier('info')).toMatchSnapshot();
    });

    test('should match createStringLiteral when literal has "-" char', () => {
      expect(createValidRouteIdentifier('user-center')).toMatchSnapshot();
    });
  });

  describe('validateIdentifierValue', () => {
    test('should return as is when string is valid', () => {
      expect(validateIdentifierValue('info')).toBe('info');
    });

    test('should return camelCase when string is invalid', () => {
      expect(validateIdentifierValue('user-center')).toBe('userCenter');
    });
  });
});
