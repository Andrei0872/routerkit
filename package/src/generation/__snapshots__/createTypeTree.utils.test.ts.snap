// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[generation] createTypeTree utils createValidRouteIdentifier should match createIdentifier 1`] = `
IdentifierObject {
  "end": -1,
  "escapedText": "info",
  "flags": 8,
  "kind": 78,
  "modifierFlagsCache": 0,
  "originalKeywordKind": undefined,
  "parent": undefined,
  "pos": -1,
  "transformFlags": 0,
}
`;

exports[`[generation] createTypeTree utils createValidRouteIdentifier should match createStringLiteral when literal has "-" char 1`] = `
TokenObject {
  "end": -1,
  "flags": 8,
  "hasExtendedUnicodeEscape": undefined,
  "kind": 10,
  "modifierFlagsCache": 0,
  "parent": undefined,
  "pos": -1,
  "singleQuote": undefined,
  "text": "user-center",
  "transformFlags": 0,
}
`;
